services/application-mapper/application-question.js

```diff
@@ -1,174 +1,120 @@
-const MentalHealthFormFields = require('../../constants/form-fields/mental-health-form-fields');
-const OffenderFormFields = require('../../constants/form-fields/offender-form-fields');
-const TreatmentFormFields = require('../../constants/form-fields/treatment-form-fields');
-const OtherCompensationFormFields = require('../../constants/form-fields/other-compensation-form-fields');
-const mapTimestampToTariffFormat = require('./date-part-mapper');
+const {DateTime} = require('luxon');
+const FormFieldsGroupedByTheme = require('../../constants/form-fields-grouped-by-theme');
 
-module.exports = class ApplicationQuestion {
-    constructor(data, oracleJson) {
-        this.value = data.value;
-        this.id = data.id;
-        this.type = data.type;
-        this.label = data.label;
-        this.sectionId = data.sectionId;
-        this.theme = data.theme;
-        this.columnName = null;
-        this.columnValue = null;
+let columnName = null;
+let columnValue = null;
 
-        this.getColumnName(this.theme);
-
-        // Check to see if id needs custom mapping
-        switch (this.id) {
-            case 'q-applicant-work-details-option':
-                if (Object.keys(oracleJson).find(column => column === this.columnName)) {
-                    const i = Object.keys(oracleJson).findIndex(
-                        column => column === this.columnName
-                    );
-                    this.columnValue = Object.values(oracleJson)[i];
-                    this.value.forEach(option => {
-                        this.columnValue = `${this.columnValue + option[0].toUpperCase()},`;
-                    });
-                    this.columnValue = this.columnValue.slice(0, -1);
-                } else {
-                    this.columnValue = '';
-                    this.value.forEach(option => {
-                        this.columnValue = `${this.columnValue + option},`;
-                    });
-                }
-                break;
-            case 'q-applicant-job-when-crime-happened':
-                if (
-                    this.value === true &&
-                    Object.keys(oracleJson).find(column => column === this.columnName)
-                ) {
-                    const i = Object.keys(oracleJson).findIndex(
-                        column => column === this.columnName
-                    );
-                    this.columnValue = `${Object.values(oracleJson)[i]}I`;
-                } else if (this.value === true) {
-                    this.columnValue = 'I,';
-                }
-                break;
-            case 'q-applicant-did-the-crime-happen-once-or-over-time':
-                if (this.value === 'once') {
-                    this.columnValue = '2';
-                } else if (this.value === 'over a period of time') {
-                    this.columnValue = '3';
+// Get the column name based off the question id
+// Form fields are grouped by theme to separate data
+function getColumnName(data) {
```

Comment:
Could we improve this by changing the format of the data? At the moment we have the column name as a key and the questionnaireId as a value, but does this need to be the case? If we flipped these round, you'd not only flatten out the data a bit, you'd also not need complex functions like this. For example if we had:
```
{
    "crime": {
        'q--whats-the-crime-reference-number': "crime_ref_no",
        'q-applicant-where-did-the-crime-happen': "incident_country",
        'q-applicant-scottish-location': 'address_line_1',
        'q-applicant-english-location': 'address_line_1',
        'q-applicant-welsh-location': 'address_line_1'
    }
}
```
Then the column name can be easily selected:
```
const columnName = FormFieldsGroupedByTheme.theme?.questionId?
```